src/witan/workspace_api.clj

```diff
@@ -1,22 +1,22 @@
 (ns witan.workspace-api
-  (:require [schema.core :as s]
-            [clojure.set]))
+  (:require [schema.core :as s]))
 
 (defn select-schema-keys
   "Like select-keys but deduces keys from a schema and performs validation"
   [schema m]
   (when-not (map? schema) (throw (Exception. "Schema must be a map")))
-  (let [has-any? (fn [x] (some #(= (type %) schema.core.AnythingSchema) x))
-        in-keys  (if (-> schema keys has-any?) [] (-> schema keys))
+  (let [has-any? (fn [x] (some #(= % s/Keyword) x))
+        in-keys  (if (-> schema keys has-any?) [] (keys schema))
         result (if (seq in-keys) (select-keys m (vec in-keys)) m)]
     (s/validate schema result)))
```

Comment:
This is quite confusing. Is this the same:

(s/validate 
  schema
  (if (has-any? (keys schema)) m (select-keys m (keys schema)))